File: src/main/java/com/gandan/practice/fibonacci/RecursiveFibonacci.java

package com.gandan.practice.fibonacci;


public class RecursiveFibonacci {
    public static void print(int n){
        for(int i=0; i<n; i++){
            System.out.println(fib(i));
        }
    }


    private static int fib(int n){
        if(n==2||n==1) {
            return 1;
        } else if(n==0){
            return 0;
        }
        return fib(n-2)+fib(n-1);
    }
}

File: src/main/java/com/gandan/practice/fibonacci/RecursiveFibonacci.kt
package com.gandan.practice.fibonacci


object RecursiveFibonacci {
    fun print(n: Int) {
        (0..n-1).map { fib(it) }.forEach { println(it) }
    }


    private fun fib(n: Int): Int {
        return when(n) {
            2, 1 -> 1
            0 -> 0
            else -> fib(n-2) + fib(n-1)
        }
    }
}
